### ISSUE TITLE
Cannot install binary using haste-cabal

### ISSUE THREAD

**Issue by kranich** [valderman/haste-compiler#317]
I cannot install `text` because its dependency `binary` fails to install with the following message
```
Resolving dependencies...
In order, the following will be installed:
binary-0.7.4.0 (reinstall) (changes: array-0.4.0.1 added, base-4.6.0.1 added, bytestring-0.10.6.0 added, containers-0.5.6.3 -> 0.4.1.0)
Warning: Note that reinstalls are always dangerous. Continuing anyway...
Configuring binary-0.7.4.0...
Building binary-0.7.4.0...
Failed to install binary-0.7.4.0
Build log ( /home/kranich/.cabal/logs/binary-0.7.4.0.log ):
Configuring binary-0.7.4.0...
Building binary-0.7.4.0...
Preprocessing library binary-0.7.4.0...

src/Data/Binary/Builder/Base.hs:192:37: Warning:
    In the use of ‘inlinePerformIO’
    (imported from Data.ByteString.Internal):
    Deprecated: "If you think you know what you are doing, use 'unsafePerformIO'. If you are sure you know what you are doing, use 'unsafeDupablePerformIO'. If you enjoy sharing an address space with a malevolent agent of chaos, try 'accursedUnutterablePerformIO'."

src/Data/Binary/Get/Internal.hs:402:21: Warning:
    In the use of ‘B.inlinePerformIO’
    (imported from Data.ByteString.Internal):
    Deprecated: "If you think you know what you are doing, use 'unsafePerformIO'. If you are sure you know what you are doing, use 'unsafeDupablePerformIO'. If you enjoy sharing an address space with a malevolent agent of chaos, try 'accursedUnutterablePerformIO'."

src/Data/Binary/Get/Internal.hs:369:1: Warning:
    Rule "<$> to <*>" may never fire because ‘<$>’ might inline first
    Probable fix: add an INLINE[n] or NOINLINE[n] pragma on ‘<$>’

src/Data/Binary/Get.hs:434:1: Warning:
    Rule "getWord8/readN" may never fire
      because ‘getWord8’ might inline first
    Probable fix: add an INLINE[n] or NOINLINE[n] pragma on ‘getWord8’

src/Data/Binary/Get.hs:435:1: Warning:
    Rule "getWord16be/readN" may never fire
      because ‘getWord16be’ might inline first
    Probable fix: add an INLINE[n] or NOINLINE[n] pragma on ‘getWord16be’

src/Data/Binary/Get.hs:436:1: Warning:
    Rule "getWord16le/readN" may never fire
      because ‘getWord16le’ might inline first
    Probable fix: add an INLINE[n] or NOINLINE[n] pragma on ‘getWord16le’

src/Data/Binary/Get.hs:437:1: Warning:
    Rule "getWord32be/readN" may never fire
      because ‘getWord32be’ might inline first
    Probable fix: add an INLINE[n] or NOINLINE[n] pragma on ‘getWord32be’

src/Data/Binary/Get.hs:438:1: Warning:
    Rule "getWord32le/readN" may never fire
      because ‘getWord32le’ might inline first
    Probable fix: add an INLINE[n] or NOINLINE[n] pragma on ‘getWord32le’

src/Data/Binary/Get.hs:439:1: Warning:
    Rule "getWord64be/readN" may never fire
      because ‘getWord64be’ might inline first
    Probable fix: add an INLINE[n] or NOINLINE[n] pragma on ‘getWord64be’

src/Data/Binary/Get.hs:440:1: Warning:
    Rule "getWord64le/readN" may never fire
      because ‘getWord64le’ might inline first
    Probable fix: add an INLINE[n] or NOINLINE[n] pragma on ‘getWord64le’
hastec: hastec: panic! (the 'impossible' happened)
  (GHC version 7.8.3 for x86_64-unknown-linux):
	cat_evals
<<details unavailable>>

Please report this as a GHC bug:  http://www.haskell.org/ghc/reportabug

cabal: Error: some packages failed to install:
binary-0.7.4.0 failed during the building phase. The exception was:
ExitFailure 1
```

**Comment by kranich** [comment]
#316 appears to be related to this issue.

**Comment by kranich** [comment]
`binary-0.7.2.3` is the latest version that can currently be installed, `binary-0.7.3.0` is the first failing version.

**Comment by TazeTSchnitzel** [comment]
This is a problem for me also: for my current project I want to use `uuid`, which depends on `binary`.

**Comment by valderman** [comment]
Fixed on master with GHC 7.10.
